x-pack/plugins/secops/public/containers/events/index.tsx

```diff
@@ -7,56 +7,105 @@
 import { getOr } from 'lodash/fp';
 import React from 'react';
 import { Query } from 'react-apollo';
+import { connect } from 'react-redux';
 import { pure } from 'recompose';
 
-import { EventItem, GetEventsQuery, KpiItem } from '../../graphql/types';
-import { inputsModel } from '../../store';
+import { Direction, Ecs, GetEventsQuery, PageInfo } from '../../graphql/types';
+import { eventsLimitSelector, inputsModel, State } from '../../store';
 import { eventsQuery } from './index.gql_query';
 
 export interface EventsArgs {
   id: string;
-  events?: EventItem[];
-  kpiEventType?: KpiItem[];
+  events: Ecs[];
   loading: boolean;
+  loadMore: (cursor: string, tiebreaker: string) => void;
+  pageInfo: PageInfo;
   refetch: inputsModel.Refetch;
+  totalCount: number;
 }
 
-export interface EventsProps {
-  id?: string;
+export interface OwnProps {
   children?: (args: EventsArgs) => React.ReactNode;
+  filterQuery?: string;
+  id?: string;
+  poll: number;
   sourceId: string;
   startDate: number;
   endDate: number;
-  filterQuery?: string;
 }
 
-export const EventsQuery = pure<EventsProps>(
-  ({ id = 'eventsQuery', children, filterQuery, sourceId, startDate, endDate }) => (
+export interface EventsComponentReduxProps {
+  limit: number;
+}
+
+type EventsProps = OwnProps & EventsComponentReduxProps;
+
+const EventsComponentQuery = pure<EventsProps>(
+  ({ children, filterQuery, id = 'eventsQuery', limit, poll, sourceId, startDate, endDate }) => (
     <Query<GetEventsQuery.Query, GetEventsQuery.Variables>
       query={eventsQuery}
       fetchPolicy="cache-and-network"
       notifyOnNetworkStatusChange
+      pollInterval={poll}
       variables={{
         filterQuery,
         sourceId,
+        pagination: {
+          limit,
+          cursor: null,
+          tiebreaker: null,
+        },
+        sortField: {
+          sortFieldId: 'timestamp',
+          direction: 'descending' as Direction,
```

Comment:
I have seen this same `as Direction` cast used in a few other places and playing with it I think that they all should be typed as:

```ts
direction: Direction.descending
```

and that will remove another `as` cast and make it to where if it's changed to a different string you won't have to hunt and change all the occurrences.